src/backend/users/management/commands/migrate_housing_seeker_data.py

```diff
@@ -0,0 +1,488 @@
+                    travel_mode=travel_mode,
+                    commute_priority=commute_priority,
+                    public_safety_priority=public_safety_priority,
+                    school_quality_priority=school_quality_priority,
+                )
+                user_profile.save()
+                self.added_profiles += 1
+        else:
+            self.skipped_user_profile_key_s3.append({"key": key, "reason": reason})
+            self.stdout.write(f"{reason}. Skipping user profile with key: {key}.\n")
+
+        return user_profile
+
+    def has_location_fields(self, destination):
+        reason = ""
+        if DestinationKeys.LOC not in destination:
+            reason += "NO LOCATION; "
+            return False, reason
+        loc = destination[DestinationKeys.LOC]
+        if LocationKeys.POSITION not in loc:
+            reason += "NO POSITION; "
+            return False, reason
+        pos = loc[LocationKeys.POSITION]
+        if PositionKeys.LAT not in pos or PositionKeys.LON not in pos:
+            reason += "NO FULL COORDINATE PAIR; "
+            return False, reason
+
+        return True, reason
+
+    def decode_purpose(self, purpose):
+        if purpose.lower() == "friends":
+            return Destination.TripPurpose.FRIENDS
+        if purpose.lower() == "work":
+            return Destination.TripPurpose.WORK
+        if purpose.lower() == "doctor":
+            return Destination.TripPurpose.DOCTOR
+        if purpose.lower() == "day care":
+            return Destination.TripPurpose.DAYCARE
+        if purpose.lower() == "family":
+            return Destination.TripPurpose.FAMILY
+        if purpose.lower() == "worship":
+            return Destination.TripPurpose.WORSHIP
+
+        return Destination.TripPurpose.OTHER
+
+    def save_destination(self, user_profile, des):
+        destination = None
+        reason = ""
+
+        has_location, has_loc_reason = self.has_location_fields(des)
+        has_purpose = self.is_key_val_nonempty(des, DestinationKeys.PURPOSE)
+        has_primary = self.is_key_val_nonempty(des, DestinationKeys.PRIMARY)
+
+        if not has_location:
+            reason += has_loc_reason
+        if not has_purpose:
+            reason += "NO PURPOSE; "
+        if not has_primary:
+            reason += "NO PRIMARY; "
+
+        # when any of the required field is empty
+        # don't attach the corresponding destination to user profile
+        if has_location and has_purpose and has_primary:
+            lat = des[DestinationKeys.LOC][LocationKeys.POSITION][PositionKeys.LAT]
+            lon = des[DestinationKeys.LOC][LocationKeys.POSITION][PositionKeys.LON]
+            purpose = self.decode_purpose(des[DestinationKeys.PURPOSE])
+
+            destination = Destination(
+                profile=user_profile,
+                location=Point(x=lat, y=lon),
+                purpose=purpose,
+                primary_destination=des[DestinationKeys.PRIMARY],
```

Comment:
Once we re-add the `address` field we'll need to update this so that it stores the `label` field into `address`. You could potentially also rename the Django field to be `label` if you coordinate with @ktohalloran on #497 .

Reply:
Ah, I see. Alright.

Hey @ktohalloran , it looks like the linked PR uses `address` as the field in the `Destination` model, and reads in the value keyed under `label`. I am adding this field back in here, and am open to either names. I think `label` would be more consistent, and may better reflect what I have seen in the user profiles on S3, since those values don't look like full street addresses to me. So let's name this field `label`- what would you say?

Reply:
That sounds good to me! I'll make updates to my PR to reflect these once this is merged.

Reply:
@ktohalloran that sounds perfect to me. Thanks!

Reply:
Update: address this here: https://github.com/azavea/echo-locator/pull/478/commits/fcfafe622990d4fb1c326ba9c21cfafcd4b5b09a